src/StrStr.java

```diff
@@ -0,0 +1,31 @@
+public class StrStr {
+    public static int strStr(String haystack, String needle) {
+        char[] haystackChars = haystack.toCharArray();
+        char[] needleChars = needle.toCharArray();
```

Comment:
Implementation is too verbose. Consider something like this:

public static int strStr(String haystack, String needle) {
        for(int i = 0; i <= haystack.length() - needle.length(); ++i) {
            boolean match = true;
            for(int j = 0; j < needle.length() && i + j < haystack.length(); j++) {
                if(haystack.charAt(i + j) != needle.charAt(j)) {
                    match = false;
                    break;
                }
            }
            if(match) {
                return i;
            }
        }
        return -1;
    }

Reply:
This solution calculates haystack.length() and needle.length() on each iteration, so instead I suggest calculating the lengths in the very beginning, storing them in haystackLength and needleLength vars. Plus, if you do charAt() on empty string - it'll throw NullpointerException, so whether this exception needs to be caught or emptiness of haystack and needle should be checked at the very beginning